rotkehlchen/chain/solana/manager.py

```diff
@@ -0,0 +1,133 @@
+    ):
+        SolanaRPCMixin.__init__(self)
+        self.greenlet_manager = greenlet_manager
+        self.database = database
+
+    def default_call_order(self) -> list[WeightedNode]:
+        """Default call order for evm nodes
+
+        Own node always has preference. Then all other node types are randomly queried
+        in sequence depending on a weighted probability.
+        """
+        open_nodes = self.database.get_rpc_nodes(
+            blockchain=SupportedBlockchain.SOLANA,
+            only_active=True,
+        )
+        selection = [wnode for wnode in open_nodes if wnode.node_info.owned is False]
+        ordered_list = []
+        while len(selection) != 0:
+            weights = [float(entry.weight) for entry in selection]
+            node = random.choices(selection, weights, k=1)
+            ordered_list.append(node[0])
+            selection.remove(node[0])
+
+        owned_nodes = [node.node_info for node in open_nodes if node.node_info.owned]
+        if len(owned_nodes) != 0:
+            # Assigning one is just a default since we always use it.
+            # The weight is only important for the other nodes since they
+            # are selected using this parameter
+            ordered_list = [WeightedNode(node_info=node, weight=ONE, active=True) for node in owned_nodes] + ordered_list  # noqa: E501
+        return ordered_list
+
+    def _query(
+            self,
+            fn: Callable[[RPCClient], R],
+            call_order: Sequence[WeightedNode],
+    ) -> R:
+        """Use the provided call order to request fn from solana"""
+        for weghted_node in call_order:
+            node_info = weghted_node.node_info
+            if (web3node := self.web3_mapping.get(node_info, None)) is None:
+                if node_info.name in self.failed_to_connect_nodes:
+                    continue
+
+                success, _ = self.attempt_connect(node=node_info)
+                if success is False:
+                    self.failed_to_connect_nodes.add(node_info.name)
+                    continue
+
+                if (web3node := self.web3_mapping.get(node_info, None)) is None:
+                    log.error(f'Unexpected missing node {node_info} at Solana')
+                    continue
+
+            try:
+                return fn(web3node.web3_instance)
+            except (HTTPError, RPCException) as e:
+                log.error(f'Failed to call {node_info.name} due to {e}')
+
+        log.error(f'Tried all solana nodes in {call_order} for {fn} but did not get any response')
+        raise RemoteError(f'Failed to get {fn}')
+
+    def get_multi_balance(self, accounts: Sequence[SolanaAddress]) -> dict[SolanaAddress, FVal]:
+        try:
+            response = self._query(
+                fn=lambda c: c.get_multiple_accounts(
+                    pubkeys=[Pubkey.from_string(addr) for addr in accounts],
+                ),
+                call_order=self.default_call_order(),
+            )
```

Comment:
why not something like?
```suggestion
            response = self._query(
                method='get_multiple_accounts',
                accounts=accounts,
                call_order=self.default_call_order(),
            )
```

then you can use overload for `self._query` to define the return type depending on the `method`